feature-libs/asm/core/facade/asm-query.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import {
+  AsmFacadeService,
+  CustomerListsPage,
+  CustomerSearchOptions,
+  CustomerSearchPage,
+} from '@spartacus/asm/root';
+import {
+  Command,
+  CommandService,
+  CommandStrategy,
+  Query,
+  QueryService,
+  QueryState,
+} from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { AsmConnector } from '../connectors/asm.connector';
+
+@Injectable()
+export class AsmQueryService implements AsmFacadeService {
+  protected customerListQuery$: Query<CustomerListsPage> =
+    this.queryService.create(() => this.asmConnector.customerLists(), {
+      reloadOn: undefined,
+      resetOn: undefined,
+    });
+
+  protected customerSearchCommand$: Command<
+    CustomerSearchOptions,
+    CustomerSearchPage
+  > = this.commandService.create(
+    (options: CustomerSearchOptions) =>
+      this.asmConnector.customerSearch(options),
+    {
+      strategy: CommandStrategy.CancelPrevious,
+    }
+  );
+
+  constructor(
+    protected queryService: QueryService,
+    protected commandService: CommandService,
+    protected asmConnector: AsmConnector
+  ) {}
+
+  getCustomerLists(): Observable<QueryState<CustomerListsPage>> {
+    return this.customerListQuery$.getState();
+  }
+
+  getCustomers(
+    options: CustomerSearchOptions = {}
+  ): Observable<CustomerSearchPage> {
+    return this.customerSearchCommand$.execute(options);
```

Comment:
I understand we use `command` because we cannot pass parameter to `query`. Can you take a look of this code, and consider whether we can use a similar way here.
```
protected searchInput$ = new Subject<{ term: string; pagination: number, ... }>();
performSearch(term: string, pagination: number, ...): void {
  this.searchInput$.next({ term, pagination, ... });
}
protected searchQuery: Query<unknown> = this.query.create(
  () =>
    this.searchInput$.pipe(
      switchMap(({ term, pagination, ... }) =>
        this.connector.search(term, pagination, ...)
      )
    )
);
```

Reply:
Can you explain more why the suggestion is more desirable?  

As I understand, the suggestion is what the Command object does.  The critical behavioral difference is that, with the suggestion, all consumers of `getCustomers()` would receive the same value.  The connector result acts like a "singleton" provider, where there can only be 1 value that is the latest page request.  This would prevents multiple consumers from having separate page states.